packages/fxa-auth-server/scripts/stripe-products-and-plans-to-firestore-documents/stripe-products-and-plans-converter.ts

```diff
@@ -0,0 +1,457 @@
+    for (const key of StyleConfigKeys) {
+      const value = stripeObject.metadata![key];
+      if (value) {
+        styleConfig[key] = value;
+      }
+    }
+    return styleConfig;
+  }
+
+  /**
+   * Convert Stripe Product or Plan support metadata keys prefixed
+   * with 'support:' into a SupportConfig object.
+   */
+  supportMetadataToSupportConfig(
+    stripeObject: Stripe.Product | Stripe.Plan
+  ): SupportConfig {
+    const supportConfig: SupportConfig = {};
+    if (!stripeObject.metadata) {
+      return supportConfig;
+    }
+    for (const key of SupportConfigKeys) {
+      const metadataKeyPrefix = `support:${key}`;
+      const metadataValues = this.getArrayOfStringsFromMetadataKeys(
+        stripeObject.metadata,
+        metadataKeyPrefix
+      );
+      if (metadataValues.length > 0) {
+        supportConfig[key] = metadataValues;
+      }
+    }
+    return supportConfig;
+  }
+
+  /**
+   * Convert Stripe Product or Plan UI Content metadata keys
+   * with and without a 'product:' prefix into a UiContentConfig
+   * object.
+   */
+  uiContentMetadataToUiContentConfig(
+    stripeObject: Stripe.Product | Stripe.Plan
+  ): UiContentConfig {
+    const uiContentConfig: UiContentConfig = {};
+    if (!stripeObject.metadata) {
+      return uiContentConfig;
+    }
+    for (const key of Object.values(UiContentConfigKeys)) {
+      let value;
+      if (key === 'details') {
+        const metadataKeyPrefix = `product:${key}`;
+        value = this.getArrayOfStringsFromMetadataKeys(
+          stripeObject.metadata,
+          metadataKeyPrefix
+        );
+        if (value.length > 0) {
+          uiContentConfig[key] = value;
+        }
+        continue;
+      }
+      value =
+        stripeObject.metadata[key] ?? stripeObject.metadata[`product:${key}`];
+      if (value) {
+        // @ts-ignore We need to be able to assign values to all keys
+        uiContentConfig[key] = value;
+      }
+    }
+    return uiContentConfig;
+  }
+
+  /**
+   * Convert Stripe Product or Plan URL metadata keys with various prefixes
+   * and suffixes to a UrlConfig object.
+   */
+  urlMetadataToUrlConfig(
+    stripeObject: Stripe.Product | Stripe.Plan
+  ): UrlConfig {
+    const urlConfig: UrlConfig = {};
+    if (!stripeObject.metadata) {
+      return urlConfig;
+    }
+    for (const key of UrlConfigKeys) {
+      const value =
+        stripeObject.metadata[`${key}URL`] ??
+        stripeObject.metadata[`${key}Link`] ??
+        stripeObject.metadata[`product:${key}URL`];
+      if (value) {
+        urlConfig[key] = value;
+      }
+    }
+    return urlConfig;
+  }
+
+  /**
+   * Convert a Stripe Product to a ProductConfig object with an empty
+   * `locales` key.
+   */
+  stripeProductToProductConfig(product: Stripe.Product): ProductConfig {
+    // Firestore cannot serialize class instances
+    const productConfig: any = {};
+
+    // BaseConfig
+    productConfig.active = product.active;
```

Comment:
Until we have a use case for 'active' property on the Firestore doc, let's hard code it to true.